### PR TITLE
Use unique socket per user for managing editor sessions

### LINKED ISSUE #6275: [Bug]: Can't start 2 instances of code-server `4.14.0` for separate users
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser: Chrome
- Local OS: Ubuntu
- Remote OS: Windows
- Remote Architecture: amd64
- `code-server --version`: 4.14.0

### Steps to Reproduce

1. Run `/usr/bin/code-server` for user 1 - ok
2. Run `/usr/bin/code-server` for user 2 - fails with the following

### Expected

Both instances should start

### Actual

2nd instance fails to start

### Logs

2nd instance tries to open the same file

[2023-06-19T16:15:12.625Z] info  code-server 4.14.0 9955cd91a4ca17e47d205e5acaf4c342a917a5e9
[2023-06-19T16:15:12.626Z] info  Using user-data-dir ~/code-server/user
[2023-06-19T16:15:12.629Z] error EPERM: operation not permitted, unlink '/tmp/code-server-ipc.sock'

### Screenshot/Video

_No response_

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

### Notes

It seems that every instance is trying to write to `/tmp/code-server-ipc.sock`, this was not the case in `4.13.0`. Is there a way to specify an alternate socket file for each instance?

### PR DESCRIPTION
**Pull request by sleexyz**
Fixes #6275

### CODE REVIEW

File: src/node/vscodeSocket.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { listen } from "./app"
 import { canConnect } from "./util"
 
 // Socket path of the daemonized code-server instance.
-export const DEFAULT_SOCKET_PATH = path.join(os.tmpdir(), "code-server-ipc.sock")
+export const DEFAULT_SOCKET_PATH = path.join(os.tmpdir(), `code-server-ipc-${os.userInfo().username}.sock`)
```
[the remark this answers is not in the document]
Comment (sleexyz) [reply src/node/vscodeSocket.ts]: Great, switched to `paths.data`.

**Review by code-asher — APPROVED** [review]

**Comment by code-asher** [comment]
I just realized we forgot to update the patch.  Maybe we can add the path the same way we add other arguments the way we pass down some other custom things like `github-auth`.

**Comment by code-asher** [comment]
Or maybe just an environment variable to make things easy.

**Comment by code-asher** [comment]
I am throwing up a PR soon with that plus a CLI flag to override.

**Comment by code-asher** [comment]
Thank you for the Jupyter example!  We changed the socket to be 
placed in `--user-data-dir` (`~/.local/share/code-server` by 
default) so it is encouraging to see other software has made 
similar decisions.

We also added a `--session-socket` flag in case someone needs to 
put it elsewhere.  Changes will be coming out in the 4.14.1 
release today.

**Pull request merged**